### PR TITLE
96413 EPS create draft appointment implementation

### PR DESCRIPTION
**Pull request by kanchanasuriya**
## Summary

- *Create draft appointment implementation*

## Related issue(s)

- *https://github.com/orgs/department-of-veterans-affairs/projects/1323/views/12?pane=issue&itemId=85928973&issue=department-of-veterans-affairs%7Cva.gov-team%7C96413*

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
NA

## Acceptance criteria

- [x]  I added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/vaos/app/services/eps/appointment_service.rb [hunk 1]
```diff
@@ -12,5 +12,16 @@ def get_appointments
...
     end
+
+    ##
+    # Create draft appointment in EPS
+    #
+    # @return OpenStruct response from EPS create draft appointment endpoint
+    #
+    def create_draft_appointment(referral_id:)
+      response = perform(:post, "/#{config.base_path}/appointments",
+                         { patientId: patient_id, referralId: referral_id }, headers)
```
Comment (randomsync) [inline modules/vaos/app/services/eps/appointment_service.rb:23]: Is referralId passed as a top level param like this correct? wellhive's POST /appointments api seems to be requiring it as a `referral` object: https://wellhive.github.io/api-docs/#/Appointment/Appointment%23create

Comment (kanchanasuriya) [reply modules/vaos/app/services/eps/appointment_service.rb:23]: Not sure if the document is outdated.  Currently, the API calls in postman works with `referralId` passed as top-level parameter.

Comment (randomsync) [reply modules/vaos/app/services/eps/appointment_service.rb:23]: ok great, yeah the api doc might be outdated. Can you also test it using rails console and make sure it works?

**Review by randomsync — APPROVED** [review]

File: modules/vaos/app/services/eps/appointment_service.rb (hunk 1 above)
Comment (kanchanasuriya) [reply modules/vaos/app/services/eps/appointment_service.rb:23]: Yeah, I verified calling this API in rails console & it works.

**Review by lee-delarm6 — APPROVED** [review]
Looking good now with the patientId change!

**Review by ericboehs — APPROVED** [review]

**Pull request merged**